integration/user_agent_test.go

```diff
@@ -0,0 +1,217 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestUserAgent(t *testing.T) {
```

Comment:
Could the three copies be consolidated? Future maintainers will need to keep reading the copies.

A single table-driven test with `optionsToAdd []string, expectedUserAgentRegexp string` entries could handle all three situations, and avoid the need for a `suite.Suite` entirely.

Alternatively, a `startMockRegistryHandler(t *testing.T) (dockerReferenceForCLI string)` (using `t.Cleanup` to terminate it), and turning `getAllUserAgents` into a `assertUserAgentMatches(t *testing.T, condition func(string) bool)` would eliminate the 3 copies while retaining the individual inputs logic in existing places.

Reply:
Oh yeah table driven testing is *way* cleaner. Sorry for not noticing that myself. I'll add something about this to my pre-review